### PR TITLE
chore(flags): adding more tests and a playground for testing out evaluation tags

### PR DESCRIPTION
**Pull request by dmarticus**
This PR adds more testing and some interactive examples for the `evaluation_environments` feature that was implemented in #2369.

### Testing

  - Added unit tests (evaluation-tags.test.ts) to verify:
    - Validation and filtering of evaluation environments
    - Proper inclusion in API requests when configured
    - Handling of edge cases (empty arrays, invalid values)
    - Backwards compatibility (no environments = no field sent)

### Examples

  - Interactive browser example (evaluation-tags-example.html):
    - Live UI for adding/removing evaluation tags
    - Real-time testing with PostHog CDN
    - Visual feedback showing how tags affect feature flag requests
    -  Try the example: `cd playground/flags && npm run serve then open http://localhost:8080/evaluation-tags-example.html`

### CODE REVIEW

File: playground/flags/evaluation-tags-example.html [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+                    config.personal_api_key = personalKey;
+                }
+
+                log('Initializing PostHog with config:', 'info');
+                log(JSON.stringify(config, null, 2), 'info');
+
+                // Initialize PostHog
+                if (window.posthog) {
+                    posthogInstance = window.posthog;
+                    posthogInstance.init(apiKey, config);
```
Comment (dustinbyrne) [inline playground/flags/evaluation-tags-example.html]: `window.posthog` will change once the client is loaded from the snippet. `posthogInstance` won't have access to data like feature flags once they're loaded. Moving this to the `loaded` callback in the config should fix this.

File: playground/flags/evaluation-tags-example.html [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+    </script>
+    
+    <!-- Option 2: Use local build (after running pnpm build) -->
+    <!-- <script src="../../packages/browser/lib/array.full.js"></script> -->
```
Comment (dustinbyrne) [inline playground/flags/evaluation-tags-example.html]:
```suggestion
    <!-- <script src="../../packages/browser/dist/array.full.js"></script> -->
```

File: playground/flags/package.json [hunk 3]
```diff
@@ -5,7 +5,9 @@
     "main": "index.js",
     "scripts": {
         "start": "node index.js",
-        "remote-config": "node remote-config-example.js"
+        "remote-config": "node remote-config-example.js",
+        "evaluation-tags": "node evaluation-tags-example.js",
+        "serve": "python3 -m http.server 8080 || python -m SimpleHTTPServer 8080"
```
Comment (dustinbyrne) [inline playground/flags/package.json:10]: This should serve the root directory if the page needs to fetch `../../packages/browser/dist/array.full.js`

**Review by dustinbyrne — APPROVED** [review]
This is cool! I had some minor issues running this locally, I've described those issues below. Tests look good.
